WordPress/gutenberg
# Remove rootTagsToEliminate prop from native Rich Text component

## What?
Removes a four-year-old temporary workaround from Aztec within the native Rich Text component.

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5549

## Why?
This prop is no longer used, and can be removed as part of an overall effort to reduce the complexity of the native Rich Text file to improve performance.

## How?
Removed the prop `rootTagsToEliminate` from the Rich Text component, and associated references from the File and Caption blocks.

## Testing Instructions
1. E2E and Writing Flow tests should be pass.
2. Within the mobile Gutenberg editor, add a file block.
3. Switch to HTML mode.
4. Note that the HTML should not contain an extra `<p>` tag.
5. Compare to the current production app, and the HTML should also not contain an extra `<p>` tag.

```
<div class="wp-block-file"><a id="wp-block-file--media-0fc3124f-4cb5-4648-9007-abbf9695f9d1" href="https://correct-humpback.jurassic.ninja/wp-content/uploads/2023/03/img_0111-2-scaled.jpg">img_0111-2-scaled.jpg</a><a href="https://correct-humpback.jurassic.ninja/wp-content/uploads/2023/03/img_0111-2-scaled.jpg" class="wp-block-file__button wp-element-button" download aria-describedby="wp-block-file--media-0fc3124f-4cb5-4648-9007-abbf9695f9d1">Download</a></div>
```

### Testing Instructions for Keyboard
This change does not affect the user interface.
